include/aspect/utilities.h

```diff
@@ -1069,6 +1069,28 @@ namespace aspect
     template <int dim>
     SymmetricTensor<2,dim> nth_basis_for_symmetric_tensors (const unsigned int k);
 
+    template <typename TestType, typename ObjectType>
+    inline
+    bool
+    object_is_of_type (const ObjectType &object)
```

Comment:
I'm not fond of that name because the real object may actually be of type *derived* from `TestType`. But I also can't come up with a better name. Want to give it another try? Maybe `object_type_matches`?